source/jormungandr/jormungandr/realtime_schedule/siri_lite.py

```diff
@@ -0,0 +1,143 @@
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from jormungandr.realtime_schedule.realtime_proxy import RealtimeProxy
+from flask import logging
+import pybreaker
+import pytz
+import requests as requests
+from jormungandr import cache, app
+from jormungandr.schedule import RealTimePassage
+from datetime import datetime
+
+
+class SiriLite(RealtimeProxy):
+    """
+    class managing calls to a siri lite external service providing real-time next passages
+    """
+    def __init__(self, id, service_url, object_id_tag=None, timeout=10, **kwargs):
+        self.service_url = service_url
+        self.timeout = timeout  # timeout in seconds
+        self.rt_system_id = id
+        self.object_id_tag = object_id_tag if object_id_tag else id
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config.get('CIRCUIT_BREAKER_MAX_SIRILITE_FAIL', 5),
+                                                reset_timeout=app.config.get('CIRCUIT_BREAKER_CLEVERAGE_TIMEOUT_S', 60))
+
+    def __repr__(self):
+        """
+         used as the cache key. we use the rt_system_id to share the cache between servers in production
+        """
+        return self.rt_system_id
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_SIRILITE', 30))
+    def _call(self, url):
+        logging.getLogger(__name__).debug('sirilite RT service, call url: {}'.format(url))
+        try:
+            return self.breaker.call(requests.get, url, timeout=self.timeout)
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('sirilite RT service dead, using base '
+                                              'schedule (error: {}'.format(e))
+            self.record_external_failure('circuit breaker open')
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('sitelite RT service timeout, using base '
+                                              'schedule (error: {}'.format(t))
+            self.record_external_failure('timeout')
+        except Exception as e:
+            logging.getLogger(__name__).exception('sirilite RT error, using base schedule')
+            self.record_external_failure(str(e))
+        return None
+
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+        line_id = route_point.fetch_line_id(self.object_id_tag)
+
+        if not stop_id or not line_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug('missing realtime id for {obj}: stop code={s}'.
+                                              format(obj=route_point, s=stop_id))
+            self.record_internal_failure('missing id')
+            return None
+
+        url = "{base_url}{stop_id}".format(base_url=self.service_url, stop_id=stop_id)
```

Comment:
actually we can't use the `lineRef` parameter, but we filter manually the response